### PR TITLE
[CARBONDATA-1762] Remove existing column level dateformat and support dateformat, timestampformat in the load option

### PR DESCRIPTION
**Pull request by akashrn5**
(1) Remove column level dateformat option
(2) Support dateformat and timestampformat in load options(table level)

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [X] Document update required?
         2 new load level properties are added. Document to be updated accordingly.
 - [X] Testing done
        UT Added
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonLoadOptionConstants.java [hunk 1]
```diff
@@ -52,6 +52,14 @@
   public static final String CARBON_OPTIONS_DATEFORMAT =
       "carbon.options.dateformat";
   public static final String CARBON_OPTIONS_DATEFORMAT_DEFAULT = "";
+
+  /**
+   * option to specify the load option
+   */
+  @CarbonProperty
+  public static final String CARBON_OPTIONS_TIMESTAMPFORMAT =
+          "carbon.options.dateformat";
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonLoadOptionConstants.java]: I think `carbon.options.dateformat ` should be `carbon.options.timestampformat`

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala [hunk 2]
```diff
@@ -17,35 +17,32 @@
...
-  def validateDateFormat(dateFormat: String, table: CarbonTable, tableName: String): Unit = {
-    val dimensions = table.getDimensionByTableName(tableName).asScala
+
+  /**
+   * validates both timestamp and date for illegal values
+   *
+   * @param optionValue
+   * @param optionName
+   */
+  def validateDateTimeFormat(optionValue: String, optionName: String): Unit = {
```
Comment (ravipesala) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala]: give proper names to parameters

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 3]
```diff
@@ -799,18 +799,19 @@ object CarbonDataRDDFactory {
...
     }
 
-    val dateFormatMap = CarbonDataProcessorUtil.getDateFormatMap(carbonLoadModel.getDateFormat)
-    val specificFormat = Option(dateFormatMap.get(partitionColumn.toLowerCase))
-    val timeStampFormat = if (specificFormat.isDefined) {
-      new SimpleDateFormat(specificFormat.get)
+    val specificTimestampFormat = carbonLoadModel.getTimestampformat
+    val specificDateFormat = carbonLoadModel.getDateFormat
+    val timeStampFormat = if (specificTimestampFormat != null &&
+                              !specificTimestampFormat.trim.isEmpty) {
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: format it properly.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1231/

**Comment by dhatchayani** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1236/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1241/

**Comment by dhatchayani** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1252/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1261/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala [hunk 4]
```diff
@@ -17,35 +17,32 @@
...
-            throw new MalformedCarbonCommandException("Error: Option DateFormat is not provided " +
-              "for " + "Column " + dateFormatSplits(0) +
-              ".")
-          }
-        }
+    if (dateTimeLoadFormat != null && dateTimeLoadFormat.trim != "") {
+      try {
+        new SimpleDateFormat(dateTimeLoadFormat)
       }
+      catch {
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala]: move to previous line

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala [hunk 5]
```diff
@@ -17,35 +17,32 @@
...
+        case _: IllegalArgumentException =>
+          throw new MalformedCarbonCommandException("Error: Wrong option: " + dateTimeLoadFormat +
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala]: use $ for variable

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 6]
```diff
@@ -799,18 +799,19 @@ object CarbonDataRDDFactory {
...
-        .CARBON_TIMESTAMP_FORMAT, CarbonCommonConstants.CARBON_TIMESTAMP_DEFAULT_FORMAT)
-      new SimpleDateFormat(timestampFormatString)
-    }
+    val specificTimestampFormat = carbonLoadModel.getTimestampformat
+    val specificDateFormat = carbonLoadModel.getDateFormat
+    val timeStampFormat =
+      if (specificTimestampFormat != null && !specificTimestampFormat.trim.isEmpty) {
+        new SimpleDateFormat(specificTimestampFormat)
+      } else {
+        val timestampFormatString = CarbonProperties.getInstance().getProperty(CarbonCommonConstants
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: move `CarbonCommonConstants` to next line, give one parameter in one line

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestLoadDataWithDiffTimestampFormat.scala [hunk 7]
```diff
@@ -75,55 +76,19 @@ class TestLoadDataWithDiffTimestampFormat extends QueryTest with BeforeAndAfterA
...
       case ex: MalformedCarbonCommandException =>
-        assertResult(ex.getMessage)("Error: Option DateFormat is not provided for Column date.")
+        assertResult(ex.getMessage)("Error: Wrong option: date is provided for option DateFormat")
       case _: Throwable=> assert(false)
     }
 
     try {
       sql(s"""
            LOAD DATA LOCAL INPATH '$resourcesPath/timeStampFormatData1.csv' into table t3
-           OPTIONS('dateformat' = 'fasfdas:yyyy/MM/dd')
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestLoadDataWithDiffTimestampFormat.scala]: why delete these testcase? `dateformat` is still a valid loading option, right?

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala [hunk 8]
```diff
@@ -17,35 +17,32 @@
...
-import org.apache.carbondata.processing.loading.model.CarbonLoadModel
 import org.apache.carbondata.processing.loading.sort.SortScopeOptions
 import org.apache.carbondata.spark.exception.MalformedCarbonCommandException
 
 object ValidateUtil {
-  def validateDateFormat(dateFormat: String, table: CarbonTable, tableName: String): Unit = {
-    val dimensions = table.getDimensionByTableName(tableName).asScala
+
+  /**
+   * validates both timestamp and date for illegal values
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala]: validate timestamp format and date format. please change

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonLoadOptionConstants.java [hunk 9]
```diff
@@ -52,6 +52,14 @@
   public static final String CARBON_OPTIONS_DATEFORMAT =
       "carbon.options.dateformat";
   public static final String CARBON_OPTIONS_DATEFORMAT_DEFAULT = "";
+
+  /**
+   * option to specify the load option
+   */
+  @CarbonProperty
+  public static final String CARBON_OPTIONS_TIMESTAMPFORMAT =
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonLoadOptionConstants.java]: please add comment for these two options to explain how it should be used: `CARBON_OPTIONS_DATEFORMAT` and `CARBON_OPTIONS_TIMESTAMPFORMAT`

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala (hunk 8 above)
Comment (dhatchayani) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/ValidateUtil.scala]: this validates both timestamp and date format

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1266/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**